ts/src/core/clause.ts

```diff
@@ -1371,6 +1371,63 @@ export function PaginationMultipleColsSubQuery<T extends Data, K = keyof T>(
   );
 }
 
+/**
+ * When paginating over multiple ordered columns, we need to construct a nested
+ * set of clauses to correctly filter. The resulting query will be structured as
+ * a group of nested ANDs and ORs.
+ *
+ * For example, given a set of ordering [{col1, less1}, {col2, greater2}], the
+ * function will construct a query that looks like:
+ *
+ * (col1 < $1 OR (col1 = $1 AND col2 > $2))
+ */
+export function PaginationUnboundColsQuery<T extends Data, K = keyof T>(
```

Comment:
Ummm, for this to work we'd then also need to have a max on the order by field list. I don't think it necessarily needs a max. It might make the cursor itself long, but I don't see it being a problem.